### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Fix a lot of clippy warnings

### CODE REVIEW

File: sdk/src/client/error.rs [hunk 1]
```diff
@@ -236,3 +236,10 @@ crate::impl_from_error_via!(ClientError via BlockError:
     UnlockError,
     SignatureError,
 );
+
+#[cfg(feature = "ledger_nano")]
+impl From<crate::client::secret::ledger_nano::Error> for ClientError {
+    fn from(value: crate::client::secret::ledger_nano::Error) -> Self {
+        Self::Ledger(value.into())
```
Comment (Thoralf-M) [inline sdk/src/client/error.rs:243]: This into() is needed?

File: sdk/src/types/block/output/account.rs [hunk 2]
```diff
@@ -563,7 +563,7 @@ fn verify_unlock_conditions(unlock_conditions: &UnlockConditions, account_id: &A
 fn verify_staking(amount: u64, features: &Features) -> Result<(), OutputError> {
     if let Some(staking) = features.staking() {
         if features.block_issuer().is_none() {
-            return Err(FeatureError::StakingBlockIssuerMissing)?;
+            return Err(FeatureError::StakingBlockIssuerMissing.into());
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/account.rs:566]: Does clippy prefer that?

File: sdk/src/client/error.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/client/error.rs:243]: to box the error

File: sdk/src/types/block/output/account.rs (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/src/types/block/output/account.rs:566]: it does

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**